### PR TITLE
Major notes update; add extensive TODO list

### PR DESCRIPTION
**Pull request by Hydreigon223**
- sscope2b, sscope2c: replace the timekeeper dump with the parent one to make it happily boot to attract
- nbapbp, nbapbpa: Add an 'IMPERFECT_CONTROLS' flag for the time being (needs JVS I/O for players 3 and 4)
- sscope/clones: Change region to UA. The region determined by the timekeeper effectively means having to add a lot of clones.

Making this a draft right now as some parts of that PR were rushed.

### CODE REVIEW

**Comment by Osso13** [comment]
Could you please stop and wait for @987123879113 to do the PR he announced? Thanks

**Review by angelosa — CHANGES_REQUESTED** [review]
Not even gonna start about what's wrong with the QA stuff.
If you are unsure about what you are talking about either remove the TODO section altogether or reword it to brief sentences.

Example:
> - terabrsta doesn't boot at all (culprit currently unknown)

Should be reworded just to

> - terabrsta: doesn't boot

File: src/mame/konami/hornet.cpp [hunk 1]
```diff
@@ -588,6 +650,11 @@ uint8_t hornet_state::sysreg_r(offs_t offset)
 			    0x04 = ADEOC (ADC EOC)
 			    0x02 = ADDOR (ADC DOR)
 			    0x01 = ADDO (ADC DO)
+			
+			FIXME: Only gradius4 and nbapbp take advantage of JVS; the rest do not.
+			In fact, the rest of the games might NOT like it if a JVS I/O board is
+			plugged in (or the SENSE line is enabled). This needs verifcation on
+			actual hardware.
```
[the remark this answers is not in the document]
Comment (Hydreigon223) [reply src/mame/konami/hornet.cpp]: If that's the case, I may as well remove this note. It's interesting Konami would leave all this leftover behavior unknown to all the cab operators or owners way back in the day when it was new.

I'm now curious if the nwk-tr games have master calendar functionality because nwk-tr boards have a JVS I/O port too.

[the remark this answers is not in the document]
Comment (987123879113) [reply src/mame/konami/hornet.cpp]: Not exactly relevant to this PR specifically but yes, all of the games in the nwk-tr driver also support the Master Calendar JVS device.

**Comment by cuavas** [comment]
You need to mention what the pull request affects in the title.

**Comment by Hydreigon223** [comment]
This PR has no further use anymore because of the progress windyfairy has made and I made an attempt to hook Silent Scope's second adc (no success so far and will explain later).

**Comment by 987123879113** [comment]
@Hydreigon223 By second adc, are you talking about the "Free" values in Silent Scope's I/O check? The ones that don't seem to function properly.

**Comment by Hydreigon223** [comment]
I was a bit late on this one. To answer your recent question, you have to set the 'skip machine init' dip off for the gun board to work.
